object-manipulation-2/src/defaults.js

```diff
@@ -0,0 +1,13 @@
+/* exported defaults */
+
+function defaults(target, source) {
+  for (var key2 in source) {
+    if (Object.keys(target).length === 0) {
+      target[key2] = source[key2];
+    } else {
+      if (!Object.keys(target).includes(key2)) {
+        target[key2] = source[key2];
```

Comment:
Normally we don't worry too much about efficiency at this point in the course, but this making the computer do a lot of extra work. You can check for the presence of a property in an object like this

```js
if (typeof target[key] === 'undefined') {
```